src/ucp/core/ucp_request.c

```diff
@@ -500,6 +509,9 @@ UCS_PROFILE_FUNC_VOID(ucp_request_memory_dereg, (context, datatype, state, req_d
 
     switch (datatype & UCP_DATATYPE_CLASS_MASK) {
     case UCP_DATATYPE_CONTIG:
+        if (req_dbg->flags & UCP_REQUEST_FLAG_USER_MEMH) {
```

Comment:
looks like now request is passed not for only debug purposes. I'd rename it to just req

Reply:
done